tests/utils.go

```diff
@@ -110,3 +120,50 @@ func GetKubeVersion(f *framework.Framework) string {
 	}
 	return ""
 }
+
+// TestNodePlacementValues returns a pre-defined set of node placement values for testing purposes.
+func TestNodePlacementValues(f *framework.Framework) cdiv1.NodePlacement {
+	nodes, _ := f.K8sClient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	affinityTestValue = v1.Affinity{
+		NodeAffinity: &v1.NodeAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+				NodeSelectorTerms: []v1.NodeSelectorTerm{
+					{
+						MatchExpressions: []v1.NodeSelectorRequirement{
+							{Key: "kubernetes.io/hostname", Operator: v1.NodeSelectorOpIn, Values: []string{nodes.Items[0].Name}},
```

Comment:
So its possible (and likely) that nodes.Items[0] is a master node. And that the master node has a taint that doesn't allow regular pods to run on them. In such an environment this will leave all the deployments with 0 pods because the node selector forces the pod to run on the node, but the taints don't allow those pods to run on the node. When you look up the nodes, filter them to exclude the no schedule taint on the nodes.

Reply:
For this reason the test is only for the existence of the particular toleration rather than it being the sole one (already)

Reply:
Am I not reading this right? This function is what generated the placement for the workloads or infra during a test for instance:
```go
cr.Spec.Workloads = tests.TestNodePlacementValues(f)
_, err := f.CdiClient.CdiV1beta1().CDIs().Update(context.TODO(), cr, metav1.UpdateOptions{})

dv := utils.NewDataVolumeWithHTTPImport("node-placement-test", "100Mi", invalidQcowLargeSize())
```
So the importer pod will use node placement information above. But as explained above, we are potentially setting it to a master node that cannot run workloads and thus the test will fail.

Reply:
I don't need the pod to actually run for the test, just have the right nodeSelector. So the test works even when it's not scheduled.

Reply:
Gotcha, still makes me a little uneasy that someone is going to use this function in a test to populate the node placement and expect it to run. Maybe just note in a comment that users of the function should not expect the pod to run.